index.bs

```diff
@@ -76,773 +93,795 @@ All <{iframe}> use cases are still valid and should continue to use <{iframe}>,
+In some environments, someone only uses a single device for a brief time to
+complete their task, like ordering in a restaurant. When their task is complete,
+the embedder application should be able to clear all of the local user data
+associated with the task and then restart the embedded instance.
 
-  :: Returns true if the current
-     <a href="https://html.spec.whatwg.org/multipage/document-sequences.html#nav-current-history-entry">
-     current session history entry</a> is not the first one in the navigation
-     history entry list. This means that there is a previous
-     <a href="https://html.spec.whatwg.org/multipage/browsing-the-web.html#session-history-entry">
-     session history entry</a> for this
-     <a href="https://html.spec.whatwg.org/multipage/document-sequences.html#navigable">
-     navigable</a>.
+<!-- ====================================================================== -->
+## Monitor for idle sessions ## {#monitor-idle-sessions}
+<!-- ====================================================================== -->
 
-  : {{ControlledFrame/reload()|reload}}()
+While users interact with embedded content, the user may not explicitly end
+their session. This content may assume the user is present when they have
+actually finished or departed without completing the task. Embedder applications
+want to detect when users idle over their case's threshold and begin a fresh
+session.
 
-  :: Reloads the current page.
+<!-- ====================================================================== -->
+## Arbitrarily blocking navigations ## {#blocking-navigations}
+<!-- ====================================================================== -->
 
-  : {{ControlledFrame/stop()|stop}}()
+While displaying embedded web content that's not authored by the embedder, pages
+may link to third party web content that's disallowed. Allowing the embedder to
+edit elements in embedded content through arbitrary script injection into the
+web content can ensure navigation cannot occur to blocked pages. The embedder
+can also use the Controlled Frame API to capture navigation events and ensure
+that only pages to approved sites can be loaded within that controlled frame.
 
-  :: Cancels the document load.
+<!-- ====================================================================== -->
+# Security, Privacy, and Accessibility Considerations # {#considerations}
+<!-- ====================================================================== -->
 
-</div>
+*This section is non-normative.*
 
 <!-- ====================================================================== -->
-## Scripting methods ## {#api-scripting}
+## Security ## {#security}
 <!-- ====================================================================== -->
 
-<xmp class="idl">
-// One of |code| or |file| must be specified but not both.
-dictionary InjectDetails {
-  DOMString code;
-  DOMString file;
-};
+<b>Controlled Frame is based upon [[Isolated-Web-Apps]] (IWA)
```

Comment:
I think it's reasonable to mention IWAs in a non-normative section like this.

Reply:
👍